### ISSUE TITLE
Building a new_gallery with a local profile or release engine fails

### ISSUE THREAD

**Issue by dnfield** [flutter/flutter#104667]
But it succeeds if you remove the deferred components.

Seems like there's something wrong with deferred component logic when using a local engine.

This makes it harder to test performance experiments in local engine builds on Android.

**Comment by GaryQian** [comment]
This failure seems to fail at the `gen_snapshot` step, @rmacnak-google

**Comment by dnfield** [comment]
I'm getting errors around it not finding the plugin related classes locally, e.g.

```
dnfield/.pub-cache/hosted/pub.dartlang.org/package_info_plus-1.4.2/android/src/main/kotlin/dev/fluttercommunity/plus/packageinfo/PackageInfoPlugin.kt: (23, 25): Unresolved reference: MethodChannel 
```

**Comment by GaryQian** [comment]
Ahh, this may be different then Ill look.

**Comment by blasten** [comment]
I'm using local engine and debug as I type, does it work if you use debug?  I used profile a week or so ago with plugins.

**Comment by dnfield** [comment]
Debug works, profile does not - but only if you have deferred app components (i.e. new_gallery)

**Comment by blasten** [comment]
To unlock this, is it possible to remove the deferred component ?

**Comment by GaryQian** [comment]
Yeah, you can build with the --no-deferred-components flag, The app will just build it into one single appbundle

**Comment by dnfield** [comment]
It does mean that contributors can't run the same benchmark as on CI though.

**Comment by rmacnak-google** [comment]
Where is new_gallery?

**Comment by guidezpl** [comment]
Related: #97149 and https://github.com/flutter/gallery/issues/926
